library/Notifications/Widget/Schedule.php

```diff
@@ -44,47 +47,59 @@ protected function assembleCalendar(Calendar $calendar): void
-                ),
-                Filter::all( // all entries that start before and end after the shown range
-                    Filter::lessThanOrEqual('start_time', $calendar->getGrid()->getGridStart()->getTimestamp()),
-                    Filter::greaterThanOrEqual('end_time', $calendar->getGrid()->getGridEnd()->getTimestamp())
-                ),
-                Filter::none( // all entries that are repeated and may still occur in the shown range
-                    Filter::lessThanOrEqual('until_time', $calendar->getGrid()->getGridStart()->getTimestamp())
-                ),
-                Filter::all( // all entries that are repeated endlessly and already started in the past
-                    Filter::unlike('until_time', '*'),
-                    Filter::like('rrule', '*'),
-                    Filter::lessThanOrEqual('start_time', $calendar->getGrid()->getGridStart()->getTimestamp())
-                )
-            );
+        $db = Database::get();
+        $entries = TimeperiodEntry::on($db)
+            ->orderBy(['start_time', 'timeperiod_id']);
+
+        $entryFilter = Filter::any(
+            Filter::all( // all entries that start in the shown range
+                Filter::greaterThanOrEqual('start_time', $calendar->getGrid()->getGridStart()->getTimestamp()),
+                Filter::lessThanOrEqual('start_time', $calendar->getGrid()->getGridEnd()->getTimestamp())
+            ),
+            Filter::all( // all entries that end in the shown range
+                Filter::greaterThanOrEqual('end_time', $calendar->getGrid()->getGridStart()->getTimestamp()),
+                Filter::lessThanOrEqual('end_time', $calendar->getGrid()->getGridEnd()->getTimestamp())
+            ),
+            Filter::all( // all entries that start before and end after the shown range
+                Filter::lessThanOrEqual('start_time', $calendar->getGrid()->getGridStart()->getTimestamp()),
+                Filter::greaterThanOrEqual('end_time', $calendar->getGrid()->getGridEnd()->getTimestamp())
+            ),
+            Filter::none( // all entries that are repeated and may still occur in the shown range
+                Filter::lessThanOrEqual('until_time', $calendar->getGrid()->getGridStart()->getTimestamp())
+            ),
+            Filter::all( // all entries that are repeated endlessly and already started in the past
+                Filter::unlike('until_time', '*'),
+                Filter::like('rrule', '*'),
+                Filter::lessThanOrEqual('start_time', $calendar->getGrid()->getGridStart()->getTimestamp())
+            )
+        );
+
+        foreach ($entries->filter($entryFilter) as $entry) {
+            $members = ScheduleMember::on($db)
+                ->with(['timeperiod', 'contact', 'contactgroup'])
+                ->filter(Filter::all(
+                    Filter::equal('schedule_id', $this->schedule->id),
+                    Filter::equal('timeperiod_id', $entry->timeperiod_id)
+                ))
```

Comment:
Now the `Filter::all` makes sense. But this way the schedule is filtered multiple times. The entries are linked to a schedule, the members also, but only indirectly. So please accept my suggestion above so that the entries are already filtered.